deployment/legacy-migration/maintenance-worker/migration/neo4j/users/users.cql

```diff
@@ -94,12 +94,11 @@
       }
     }
 */
-
 CALL apoc.load.json("file:${IMPORT_CHUNK_PATH_CQL_FILE}") YIELD value as user
 MERGE(u:User {id: user._id["$oid"]})
 ON CREATE SET
 u.name       = user.name,
-u.slug       = user.slug,
+u.slug       = COALESCE(user.slug, apoc.text.random(20, "[A-Za-z]")),
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-16T07:22:35Z" title="Friday, August 16th 2019, 9:22:35 am +02:00">Aug 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -94,12 +94,11 @@
       }
     }
 */
-
 CALL apoc.load.json("file:${IMPORT_CHUNK_PATH_CQL_FILE}") YIELD value as user
 MERGE(u:User {id: user._id["$oid"]})
 ON CREATE SET
 u.name       = user.name,
-u.slug       = user.slug,
+u.slug       = COALESCE(user.slug, apoc.text.random(20, "[A-Za-z]")),
```


from the docs `coalesce() returns the first non-null value in the given list of expressions.`
since `apoc.text.random()` will never return null, this is a safer way to import the data! 
Nice @roschaefer

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-16T07:25:52Z" title="Friday, August 16th 2019, 9:25:52 am +02:00">Aug 16, 2019</time>_

---


one thing I didn't notice though @roschaefer is that you are using `A-Z`, shouldn't we avoid using slug with capital letters?
I already merged it, but we could refactor, or you could explain why it's fine...

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-16T08:33:51Z" title="Friday, August 16th 2019, 10:33:51 am +02:00">Aug 16, 2019</time>_

---


@mattwr18 oh, yes you're right. We could also prefix the slug with `unknown-`... so these users show up at the end of the list.